components/script/dom/fontfaceset.rs

```diff
@@ -24,19 +24,24 @@ pub struct FontFaceSet {
 }
 
 impl FontFaceSet {
-    pub fn new_inherited(global: &GlobalScope, can_gc: CanGc) -> Self {
-        FontFaceSet {
+    pub fn new_inherited(global: &GlobalScope, is_ready: bool) -> Self {
+        let promise = Promise::new(global, CanGc::note());
+        if is_ready {
+            promise.resolve_native(&promise);
```

Comment:
This doesn't match the behavior of fulfill_ready_promise_if_needed, which resolves the promise with the FontFaceSet self value.

Reply:
We will need to move the resolve logic into new instead of new_inherired, so that we have a valid instance of FontFaceSet to pass.